From 9ce80fdc36424fede0e129bed06ef6d64a88bb5c Mon Sep 17 00:00:00 2001
From: Konstantin I Boudnik <cos@apache.org>
Date: Fri, 5 Apr 2013 20:45:13 +0000
Subject: [PATCH 1328/1357] HDFS-4646. createNNProxyWithClientProtocol ignores configured timeout value.
 Contributed by Jagane Sundar

git-svn-id: https://svn.apache.org/repos/asf/hadoop/common/branches/branch-2@1465123 13f79535-47bb-0310-9956-ffa450edef68
(cherry picked from commit 56d8bf64459fc33f7cfc45de72d62dff8a4cef6e)
---
 .../org/apache/hadoop/hdfs/NameNodeProxies.java    |    4 +++-
 1 files changed, 3 insertions(+), 1 deletions(-)

diff --git a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/NameNodeProxies.java b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/NameNodeProxies.java
index 1f582eb..82de952 100644
--- a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/NameNodeProxies.java
+++ b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/NameNodeProxies.java
@@ -346,7 +346,9 @@ public class NameNodeProxies {
     final long version = RPC.getProtocolVersion(ClientNamenodeProtocolPB.class);
     ClientNamenodeProtocolPB proxy = RPC.getProtocolProxy(
         ClientNamenodeProtocolPB.class, version, address, ugi, conf,
-        NetUtils.getDefaultSocketFactory(conf), 0, defaultPolicy).getProxy();
+        NetUtils.getDefaultSocketFactory(conf),
+        org.apache.hadoop.ipc.Client.getTimeout(conf), defaultPolicy)
+            .getProxy();
 
     if (withRetries) { // create the proxy with retries
 
-- 
1.7.0.4

